arrow/table_buffer.go

```diff
@@ -83,11 +83,41 @@ func (t *TableBuffer) Validate() error {
 		return nil
 	}
 
-	sz := t.Values[0].Len()
-	for i := 1; i < len(t.Values); i++ {
-		if t.Values[i].Len() != sz {
+	var sz int
```

Comment:
This variable is initialized on the first iteration of this loop and is used in subsequent loops. If I put it in the for loop, it will not keep the size for subsequent loops.

I could initialize this once outside of the for loop instead of inside of the for loop, but then I'd have to add an additional check for nil because it's possible for the first column to be nil.